apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/FissClaimResponseTransformerV2.java

```diff
@@ -88,50 +85,129 @@ private static ClaimResponse transformClaim(PreAdjFissClaim claimGroup) {
+                                    CommonCodings.MC.getCode(),
+                                    CommonCodings.MC.getDisplay())))
+                        .setSystem(
+                            TransformerConstants.CODING_BBAPI_MEDICARE_BENEFICIARY_ID_UNHASHED)
+                        .setValue(claimGroup.getMbi())));
+    patient.setId("patient");
+
+    if (optional.isPresent()) {
+      PreAdjFissPayer benePayer = optional.get();
+      patient
+          .setName(getBeneName(benePayer))
+          .setBirthDate(localDateToDate(benePayer.getBeneDob()))
+          .setGender(
+              benePayer.getBeneSex() == null
+                  ? null
+                  : GENDER_MAP.get(benePayer.getBeneSex().toLowerCase()));
+    }
+
+    return patient;
+  }
+
+  private static List<HumanName> getBeneName(PreAdjFissPayer benePayer) {
+    List<HumanName> names;
+
+    if (benePayer.getBeneLastName() != null
+        || benePayer.getBeneFirstName() != null
+        || benePayer.getBeneMidInit() != null) {
+      HumanName name = new HumanName();
+
+      if (benePayer.getBeneLastName() != null) {
+        name.setFamily(benePayer.getBeneLastName());
+      }
+
+      if (benePayer.getBeneFirstName() != null || benePayer.getBeneMidInit() != null) {
+        name.setGiven(
+            List.of(
+                new StringType(benePayer.getBeneFirstName()),
+                new StringType(
+                    benePayer.getBeneMidInit() == null ? null : benePayer.getBeneMidInit() + ".")));
+      }
+
+      names = List.of(name);
+    } else {
+      names = null;
+    }
+
+    return names;
+  }
+
   private static List<Extension> getExtension(PreAdjFissClaim claimGroup) {
-    return Collections.singletonList(
-        new Extension()
-            .setUrl("https://dcgeo.cms.gov/resources/variables/fiss-status")
+    List<Extension> extensions = new ArrayList<>();
+    extensions.add(
+        new Extension(BBCodingSystems.FISS.CURR_STATUS)
             .setValue(
                 new Coding(
-                    "https://dcgeo.cms.gov/resources/variables/fiss-status",
-                    "" + claimGroup.getCurrStatus(),
-                    STATUS_TEXT.get(Character.toLowerCase(claimGroup.getCurrStatus())))));
+                    BBCodingSystems.FISS.CURR_STATUS, "" + claimGroup.getCurrStatus(), null)));
+
+    if (claimGroup.getReceivedDate() != null) {
+      extensions.add(
+          new Extension(BBCodingSystems.FISS.RECD_DT_CYMD)
+              .setValue(new DateType(localDateToDate(claimGroup.getReceivedDate()))));
+    }
+
+    if (claimGroup.getCurrTranDate() != null) {
+      extensions.add(new Extension(BBCodingSystems.FISS.CURR_TRAN_DT_CYMD));
+    }
+
+    return List.copyOf(extensions);
   }
 
   private static List<Identifier> getIdentifier(PreAdjFissClaim claimGroup) {
-    return Collections.singletonList(
+    return List.of(
         new Identifier()
-            .setType(
```

Comment:
actually there was a mistake in the JSON, we should keep the C4BBIdentifierType coding